Libraries/LibWeb/HTML/FormAssociatedElement.cpp

```diff
@@ -248,6 +249,36 @@ bool FormAssociatedElement::check_validity_steps()
     return true;
 }
 
+// https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#report-validity-steps
+bool FormAssociatedElement::report_validity_steps()
+{
+    // 1. If element is a candidate for constraint validation and does not satisfy its constraints, then:
+    if (is_candidate_for_constraint_validation() && !satisfies_its_constraints()) {
+        auto& element = form_associated_element_to_html_element();
+        // 1. Let report be the result of firing an event named invalid at element, with the cancelable attribute initialized to true.
+        auto report = element.dispatch_event(DOM::Event::create(element.realm(), EventNames::invalid, { .cancelable = true }));
+
+        // FIXME: 2. If report is true, then report the problems with the constraints of this element to the user. When reporting the problem with the constraints to the user,
```

Comment:
I feel like the FIXME is kinda inappropriate here here. (and in the other function)
As it is, this implementation is to spec, so I think the FIXMEs should be where you have "What exactly should happen here?" and instead say that we might want to give better indication to the user or align with other browsers.

The spec feels intentionally vague here so I'm not sure if this needs to be exactly interoperable with other browsers.
I'd expect the spec to state clearer requirements if that were the case.

Reply:
I have adapted the comments. 

Other browsers also show the type of validation error. However, I have not checked exactly what happens in case of multiple errors or multiple incorrect inputs.